File: Striver/Recursion/q3.java
package Striver.Recursion;

public class q3 {
    // program to print sum of all the numbers from 1 to n using recursion
    public static void main(String[] args) {
        int n = 5;
        sum(n, 0);
    }

    static void sum(int i, int sum) {
        if (i < 1) {
            System.out.println(sum);
            return;
        }
        sum += i;
        sum(i - 1, sum);
    }
}

File: Striver/Recursion/q3.py
# program to print sum of all the numbers from 1 to n using recursion

def sumNum(i,sum):
    if i<1:
        print(sum)
        return
    sum = sum+i
    sumNum(i-1,sum)

n = int(input("Enter the number: "))
sumNum(n,0)